src/OpenFeature.DependencyInjection/OpenFeatureBuilderExtensions.cs

```diff
@@ -49,17 +52,216 @@ public static OpenFeatureBuilder AddContext(
+    public static OpenFeatureBuilder AddProvider<TOptions, TProviderFactory>(this OpenFeatureBuilder builder, string name, Action<TProviderFactory>? configureFactory = null)
+        where TOptions : OpenFeatureOptions, new()
+        where TProviderFactory : class, IFeatureProviderFactory, new()
+    {
+        Guard.ThrowIfNullOrWhiteSpace(name, nameof(name));
+
+        builder.NamedProviderRegistrationCount++;
+
+        builder.Services.Configure<TOptions>(options =>
+        {
+            options.AddProviderName(name);
+        });
+
+        if (configureFactory != null)
+        {
+            builder.Services.AddOptions<TProviderFactory>(name)
+                .Validate(options => options != null, $"{typeof(TProviderFactory).Name} configuration is invalid.")
+                .Configure(configureFactory);
+        }
+        else
+        {
+            builder.Services.AddOptions<TProviderFactory>(name)
+                .Configure(options => { });
+        }
+
+        builder.Services.TryAddKeyedSingleton(name, static (provider, key) =>
+        {
+            var options = provider.GetRequiredService<IOptionsMonitor<TProviderFactory>>();
+            var providerFactory = options.Get(key!.ToString());
+            return providerFactory.Create();
+        });
+
+        builder.AddClient(name);
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a named feature provider with a specified configuration builder, using default <see cref="OpenFeatureOptions"/>.
+    /// </summary>
+    /// <typeparam name="TProviderFactory">The type of the provider factory implementing <see cref="IFeatureProviderFactory"/>.</typeparam>
+    /// <param name="builder">The <see cref="OpenFeatureBuilder"/> instance.</param>
+    /// <param name="name">The unique name of the provider.</param>
+    /// <param name="configureFactory">An optional action to configure the provider factory of type <typeparamref name="TProviderFactory"/>.</param>
+    /// <returns>The configured <see cref="OpenFeatureBuilder"/> instance.</returns>
+    public static OpenFeatureBuilder AddProvider<TProviderFactory>(this OpenFeatureBuilder builder, string name, Action<TProviderFactory>? configureFactory = null)
+        where TProviderFactory : class, IFeatureProviderFactory, new()
+        => AddProvider<OpenFeatureOptions, TProviderFactory>(builder, name, configureFactory);
+
+    /// <summary>
+    /// Adds a feature client to the service collection, configuring it to work with a specific context if provided.
+    /// </summary>
+    /// <param name="builder">The <see cref="OpenFeatureBuilder"/> instance.</param>
+    /// <param name="name">Optional: The name for the feature client instance.</param>
+    /// <returns>The <see cref="OpenFeatureBuilder"/> instance.</returns>
+    internal static OpenFeatureBuilder AddClient(this OpenFeatureBuilder builder, string? name = null)
+    {
+        if (name == null)
```

Comment:
Does it make sense to check for empty or white space?

Reply:
fixed